base_layer/core/src/mining/miner.rs

```diff
@@ -0,0 +1,137 @@
+// Copyright 2019. The Tari Project
+//
+// Redistribution and use in source and binary forms, with or without modification, are permitted provided that the
+// following conditions are met:
+//
+// 1. Redistributions of source code must retain the above copyright notice, this list of conditions and the following
+// disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the
+// following disclaimer in the documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse or promote
+// products derived from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+// INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
+
+// USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+use crate::{
+    blocks::{genesis_block::get_genesis_block, Block, NewBlockTemplate},
+    chain_storage::BlockchainBackend,
+    consensus::{ConsensusConstants, ConsensusManager},
+    mining::{blake_miner::CpuBlakePow, error::MinerError, CoinbaseBuilder},
+    proof_of_work::Difficulty,
+};
+use core::sync::atomic::AtomicBool;
+use futures::channel::mpsc::{Receiver, Sender};
+
+use std::sync::{atomic::Ordering, Arc};
+use tari_crypto::keys::SecretKey;
+use tari_transactions::{
+    tari_amount::MicroTari,
+    transaction::Transaction,
+    types::{CryptoFactories, PrivateKey},
+};
+
+struct Miner<B: BlockchainBackend> {
```

Comment:
I wonder if this Miner shouldn't run in a MinerService that uses the current service framework. This will make it easier to spawn the miner, get a handle to the the BaseNodeService and shut down properly when a shutdown signal was sent.

Reply:
The miner is basically going to get jettisoned at mainnet, so I wouldn't spend any time more than the minimum on this

Reply:
I see, then a miner service is overkill.